assets/src/domain/eventEditor/ui/datetimes/dateForm/useDateFormConfig.ts

```diff
@@ -0,0 +1,126 @@
+import { __ } from '@wordpress/i18n';
+import { format, parseISO, parse } from 'date-fns';
+
+import { EspressoFormProps, dateFormat, timeFormat } from '@application/ui/forms/espressoForm';
+import useDatetimeItem from '../../../services/apollo/queries/datetimes/useDatetimeItem';
+import { EntityId } from '@edtrServices/apollo/types';
+import { PLUS_ONE_MONTH, PLUS_TWO_MONTHS } from '../../../../shared/constants/defaultDates';
+
+interface DateFormShape {
+	name?: string;
+	description?: string;
+	capacity?: number;
+	dateTime?: {
+		startDate: string;
+		startTime: string;
+		endDate: string;
+		endTime: string;
+	};
+	isTrashed?: boolean;
+}
+
+type DateFormConfig = EspressoFormProps<DateFormShape>;
+
+const useDateFormConfig = (id: EntityId, config?: EspressoFormProps): DateFormConfig => {
+	const { name, description, capacity, startDate: start, endDate: end } = useDatetimeItem({ id }) || {};
+
+	const startDate = start ? parseISO(start) : PLUS_ONE_MONTH;
+	const endDate = end ? parseISO(end) : PLUS_TWO_MONTHS;
+
+	const { onSubmit } = config;
+
+	const onSubmitFrom: DateFormConfig['onSubmit'] = ({ dateTime, ...rest }, form, ...restParams) => {
+		// convert "dateTime" object to proper "startDate" and "endDate"
+		const startDateStr = `${dateTime.startDate} ${dateTime.startTime}`;
+		const endDateStr = `${dateTime.endDate} ${dateTime.endTime}`;
+		const formatStr = `${dateFormat} ${timeFormat}`;
```

Comment:
perhaps, in the future, this logic might be extracted into some kind of utility functions and added some tests